Riello IDG400 wrong values & not powering off

Hello,
I just changed my old UPS with a new Riello IDG400:
1: it shows battery charge 255% - battery.runtime 3932100 - Runtime (H:M:S): 1092:15:00
is this an hardware problem or something on Nut side?
2: I use a raspberry as nut server to shutdown all lan clients after 180 seconds without main power; the old ups was also powered off after FINALDELAY 5 seconds while the Riello keeps staying on battery; is this something related to the first problem?
Thanks

@clepple 

make still not working 

sequence of events as follows..

login as: pi
pi@192.168.1.239's password:
Linux raspberrypi 4.19.57+ #1244 Thu Jul 4 18:42:50 BST 2019 armv6l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.

SSH is enabled and the default password for the 'pi' user has not been changed.
This is a security risk - please login as the 'pi' user and type 'passwd' to set                                                                                         a new password.

pi@raspberrypi:~ $ sudo apt-get update
Get:1 http://archive.raspberrypi.org/debian buster InRelease [25.1 kB]
Get:2 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]
Get:3 http://raspbian.raspberrypi.org/raspbian buster/main armhf Packages [13.0 MB]
Get:4 http://archive.raspberrypi.org/debian buster/main armhf Packages [205 kB]
Fetched 13.2 MB in 39s (339 kB/s)
Reading package lists... Done
pi@raspberrypi:~ $ sudo apt-get build-dep nut
Reading package lists... Done
E: You must put some 'source' URIs in your sources.list
pi@raspberrypi:~ $ sudo nano  /etc/apt/source.list
pi@raspberrypi:~ $ sudo nano  /etc/apt/sources
pi@raspberrypi:~ $ ls
pi@raspberrypi:~ $ ls
pi@raspberrypi:~ $ ls -l
total 0
pi@raspberrypi:~ $ sudo apt-get -y install git
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  git-man libcurl3-gnutls liberror-perl
Suggested packages:
  git-daemon-run | git-daemon-sysvinit git-doc git-el git-email git-gui gitk gitweb git-cvs git-mediawiki git-svn
The following NEW packages will be installed:
  git git-man libcurl3-gnutls liberror-perl
0 upgraded, 4 newly installed, 0 to remove and 0 not upgraded.
Need to get 6,144 kB of archives.
After this operation, 32.8 MB of additional disk space will be used.
Get:1 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf libcurl3-gnutls armhf 7.64.0-4 [291 kB]
Get:2 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf liberror-perl all 0.17027-2 [30.9 kB]
Get:3 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf git-man all 1:2.20.1-2 [1,619 kB]
Get:4 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf git armhf 1:2.20.1-2 [4,202 kB]
Fetched 6,144 kB in 3s (2,207 kB/s)
Selecting previously unselected package libcurl3-gnutls:armhf.
(Reading database ... 37620 files and directories currently installed.)
Preparing to unpack .../libcurl3-gnutls_7.64.0-4_armhf.deb ...
Unpacking libcurl3-gnutls:armhf (7.64.0-4) ...
Selecting previously unselected package liberror-perl.
Preparing to unpack .../liberror-perl_0.17027-2_all.deb ...
Unpacking liberror-perl (0.17027-2) ...
Selecting previously unselected package git-man.
Preparing to unpack .../git-man_1%3a2.20.1-2_all.deb ...
Unpacking git-man (1:2.20.1-2) ...
Selecting previously unselected package git.
Preparing to unpack .../git_1%3a2.20.1-2_armhf.deb ...
Unpacking git (1:2.20.1-2) ...
Setting up libcurl3-gnutls:armhf (7.64.0-4) ...
Setting up liberror-perl (0.17027-2) ...
Setting up git-man (1:2.20.1-2) ...
Setting up git (1:2.20.1-2) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for libc-bin (2.28-10+rpi1) ...
pi@raspberrypi:~ $ git clone https://github.com/networkupstools/nut.git
Cloning into 'nut'...
remote: Enumerating objects: 54, done.
remote: Counting objects: 100% (54/54), done.
remote: Compressing objects: 100% (50/50), done.
remote: Total 40384 (delta 14), reused 16 (delta 4), pack-reused 40330
Receiving objects: 100% (40384/40384), 67.82 MiB | 835.00 KiB/s, done.
Resolving deltas: 100% (29594/29594), done.
Checking out files: 100% (706/706), done.
pi@raspberrypi:~ $ zcat patch/attachment.bin | patch -p0
gzip: patch/attachment.bin.gz: No such file or directory
pi@raspberrypi:~ $ ls
nut
pi@raspberrypi:~ $ cd nut
pi@raspberrypi:~/nut $ zcat patch/attachment.bin | patch -p0
patching file drivers/riello_ser.c
patching file drivers/riello_usb.c
pi@raspberrypi:~/nut $ ./configure --includedir=/usr/include --mandir=/usr/share/man \
> --infodir=/usr/share/info --sysconfdir=/etc/nut --localstatedir=/var \
> --libexecdir=/usr/lib/nut --srcdir=. --enable-maintainer-mode \
> --disable-silent-rules --libdir=/usr/lib/arm-linux-gnueabihf \
> --with-ssl --with-nss --with-cgi --with-dev --enable-static \
> --with-statepath=/var/run/nut --with-altpidpath=/var/run/nut \
> --with-drvpath=/lib/nut --with-cgipath=/usr/lib/cgi-bin/nut \
> --with-htmlpath=/usr/share/nut/www --with-pidpath=/var/run/nut \
> --datadir=/usr/share/nut --with-pkgconfig-dir=/usr/lib/arm-linux-gnueabihf/pkgconfig \
> --with-user=nut --with-group=nut --with-udev-dir=/lib/udev \
> --with-systemdsystemunitdir=/lib/systemd/system
-bash: ./configure: No such file or directory
pi@raspberrypi:~/nut $ ./autogen.sh
Regenerating Augeas ups.conf lens...
Regenerating the USB helper files...
Calling autoreconf...
./autogen.sh: 44: ./autogen.sh: autoreconf: not found
pi@raspberrypi:~/nut $ sudo apt-get install autoconf
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  automake autotools-dev libsigsegv2 m4
Suggested packages:
  autoconf-archive gnu-standards autoconf-doc libtool gettext m4-doc
The following NEW packages will be installed:
  autoconf automake autotools-dev libsigsegv2 m4
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 1,407 kB of archives.
After this operation, 4,276 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf libsigsegv2 armhf 2.12-2 [32.3 kB]
Get:2 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf m4 armhf 1.4.18-2 [185 kB]
Get:3 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf autoconf all 2.69-11 [341 kB]
Get:4 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf autotools-dev all 20180224.1 [77.0 kB]
Get:5 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf automake all 1:1.16.1-4 [771 kB]
Fetched 1,407 kB in 1s (1,137 kB/s)
Selecting previously unselected package libsigsegv2:armhf.
(Reading database ... 38556 files and directories currently installed.)
Preparing to unpack .../libsigsegv2_2.12-2_armhf.deb ...
Unpacking libsigsegv2:armhf (2.12-2) ...
Selecting previously unselected package m4.
Preparing to unpack .../archives/m4_1.4.18-2_armhf.deb ...
Unpacking m4 (1.4.18-2) ...
Selecting previously unselected package autoconf.
Preparing to unpack .../autoconf_2.69-11_all.deb ...
Unpacking autoconf (2.69-11) ...
Selecting previously unselected package autotools-dev.
Preparing to unpack .../autotools-dev_20180224.1_all.deb ...
Unpacking autotools-dev (20180224.1) ...
Selecting previously unselected package automake.
Preparing to unpack .../automake_1%3a1.16.1-4_all.deb ...
Unpacking automake (1:1.16.1-4) ...
Setting up autotools-dev (20180224.1) ...
Setting up libsigsegv2:armhf (2.12-2) ...
Setting up m4 (1.4.18-2) ...
Setting up autoconf (2.69-11) ...
Setting up automake (1:1.16.1-4) ...
update-alternatives: using /usr/bin/automake-1.16 to provide /usr/bin/automake (automake) in auto mode
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for install-info (6.5.0.dfsg.1-4+b1) ...
Processing triggers for libc-bin (2.28-10+rpi1) ...
pi@raspberrypi:~/nut $ ./autogen.sh
Calling autoreconf...
configure.ac:135: error: possibly undefined macro: AC_PROG_LIBTOOL
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:904: error: possibly undefined macro: AC_DISABLE_STATIC
autoreconf: /usr/bin/autoconf failed with exit status: 1
pi@raspberrypi:~/nut $ sudo apt-get install libtool
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  libltdl-dev libltdl7
Suggested packages:
  libtool-doc gfortran | fortran95-compiler gcj-jdk
The following NEW packages will be installed:
  libltdl-dev libltdl7 libtool
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 1,093 kB of archives.
After this operation, 2,606 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf libltdl7 armhf 2.4.6-9 [387 kB]
Get:2 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf libltdl-dev armhf 2.4.6-9 [159 kB]
Get:3 http://ftp.heanet.ie/mirrors/raspbian/raspbian buster/main armhf libtool all 2.4.6-9 [547 kB]
Fetched 1,093 kB in 1s (1,162 kB/s)
Selecting previously unselected package libltdl7:armhf.
(Reading database ... 38851 files and directories currently installed.)
Preparing to unpack .../libltdl7_2.4.6-9_armhf.deb ...
Unpacking libltdl7:armhf (2.4.6-9) ...
Selecting previously unselected package libltdl-dev:armhf.
Preparing to unpack .../libltdl-dev_2.4.6-9_armhf.deb ...
Unpacking libltdl-dev:armhf (2.4.6-9) ...
Selecting previously unselected package libtool.
Preparing to unpack .../libtool_2.4.6-9_all.deb ...
Unpacking libtool (2.4.6-9) ...
Setting up libtool (2.4.6-9) ...
Setting up libltdl7:armhf (2.4.6-9) ...
Setting up libltdl-dev:armhf (2.4.6-9) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for libc-bin (2.28-10+rpi1) ...
pi@raspberrypi:~/nut $ ./autogen.sh
Calling autoreconf...
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:11: installing './config.guess'
configure.ac:11: installing './config.sub'
configure.ac:15: installing './install-sh'
configure.ac:15: installing './missing'
Makefile.am: installing './INSTALL'
clients/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
pi@raspberrypi:~/nut $ ./configure --includedir=/usr/include --mandir=/usr/share/man \
> --infodir=/usr/share/info --sysconfdir=/etc/nut --localstatedir=/var \
> --libexecdir=/usr/lib/nut --srcdir=. --enable-maintainer-mode \
> --disable-silent-rules --libdir=/usr/lib/`gcc -print-multiarch` \
> --with-ssl --with-nss --with-cgi --with-dev --enable-static \
> --with-statepath=/var/run/nut --with-altpidpath=/var/run/nut \
> --with-drvpath=/lib/nut --with-cgipath=/usr/lib/cgi-bin/nut \
> --with-htmlpath=/usr/share/nut/www --with-pidpath=/var/run/nut \
> --datadir=/usr/share/nut --with-pkgconfig-dir=/usr/lib/`gcc -print-multiarch`/pkgconfig \
> --with-user=nut --with-group=nut --with-udev-dir=/lib/udev \
> --with-systemdsystemunitdir=/lib/systemd/system
Network UPS Tools version 2.7.4.1
checking build system type... armv6l-unknown-linux-gnueabihf
checking host system type... armv6l-unknown-linux-gnueabihf
checking target system type... armv6l-unknown-linux-gnueabihf
checking for host system name... debian
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for autoconf macro to enable system extensions... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking for pkg-config... /usr/bin/pkg-config
configure: checking for autoconf macro support of pkg-config
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
configure: checking for autoconf macro support of pkg-config module checker
checking for dummy_PKG_CONFIG... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking whether g++ understands -c and -o together... yes
checking whether ln -s works... yes
checking for egrep... (cached) /bin/grep -E
checking for ar... /usr/bin/ar
checking for ranlib... ranlib
checking whether byte ordering is bigendian... no
checking for inline... inline
checking for flexible array members... yes
checking for variable-length arrays... yes
checking for flock... yes
checking for lockf... yes
checking for fcvt... yes
checking for fcvtl... no
checking for pow10... no
checking for round... no
checking for abs_val... no
checking for cfsetispeed... yes
checking for tcsendbreak... yes
checking for seteuid... yes
checking for setsid... yes
checking for getpassphrase... no
checking for on_exit... yes
checking for strptime... yes
checking for setlogmask... yes
checking whether LOG_UPTO is declared... yes
checking whether ln -sr works... yes
checking for unsigned long long int... yes
checking for vsnprintf... yes
checking for snprintf... yes
checking for setenv... yes
checking for strerror... yes
checking for atexit... yes
checking whether optind is declared... yes
checking for getopt.h... yes
checking for getopt_long... yes
checking whether uu_lock is declared... no
checking whether __func__ is declared... yes
checking for library containing gethostbyname... none required
checking for library containing connect... none required
checking whether time.h and sys/time.h may both be included... yes
checking for sys/modem.h... no
checking for stdarg.h... yes
checking for varargs.h... no
checking for sys/termios.h... yes
checking for sys/time.h... yes
checking for library containing pthread_create... -lpthread
checking for socklen_t... yes
checking for --with-all... not given
checking for Net-SNMP version via net-snmp-config... none found
checking for Net-SNMP cflags...
checking for Net-SNMP libs...
checking for net-snmp/net-snmp-config.h... no
checking for init_snmp... no
checking for libusb version via pkg-config... checking via libusb-config... none found
checking for libusb cflags...
checking for libusb ldflags... -lusb
checking for usb.h... no
checking for usb_init... no
checking for libneon version via pkg-config (0.25.0 minimum required)... none found
configure: WARNING: could not get pkg-config information for libneon version, using fallback defaults
checking for libneon cflags... -I/usr/include/neon
configure: WARNING: could not get pkg-config information for libneon cflags, using fallback defaults
checking for libneon ldflags... -lneon
configure: WARNING: could not get pkg-config information for libneon libs, using fallback defaults
checking for ne_xmlreq.h... no
checking for ne_xml_dispatch_request... no
checking for libpowerman cflags...
checking for libpowerman libs...
checking for libpowerman.h... no
checking for pm_connect... no
checking for libmodbus version via pkg-config... none found
checking for libmodbus cflags... -I/usr/include/modbus
checking for libmodbus ldflags... -lmodbus
checking for modbus.h... no
checking for modbus_new_rtu... no
checking for modbus_new_tcp... no
checking for avahi-core version via pkg-config (0.6.30 minimum required)... none found
checking for avahi-client version via pkg-config (0.6.30 minimum required)... none found
checking for avahi cflags...
checking for avahi ldflags...
checking for avahi-common/malloc.h... no
checking for avahi_free... no
checking whether to build serial drivers... yes
checking whether to build USB drivers... no
checking whether to build neon based XML driver... no
checking whether to enable Avahi support... no
checking whether to build Powerman PDU client driver... no
checking whether to build Modbus driver... no
checking for FreeIPMI version via pkg-config... none found
checking for FreeIPMI cflags...
checking for FreeIPMI ldflags... -lfreeipmi -lipmimonitoring
checking for freeipmi/freeipmi.h... no
checking for ipmi_monitoring.h... no
checking for library containing ipmi_ctx_create... no
checking for library containing ipmi_ctx_find_inband... no
checking for library containing ipmi_monitoring_init... no
checking for library containing ipmi_monitoring_sensor_read_record_id... no
checking for library containing ipmi_sdr_cache_ctx_destroy... no
checking for library containing ipmi_sdr_ctx_destroy... no
checking whether to build IPMI driver... no
checking whether to build Mac OS X meta-driver... no
checking whether i2c_smbus_read_word_data is declared... no
checking whether i2c_smbus_write_word_data is declared... no
checking whether i2c_smbus_read_block_data is declared... no
checking whether to build i2c based drivers... no
checking for Mozilla NSS version via pkg-config... none found
checking for Mozilla NSS cflags...
checking for Mozilla NSS ldflags... -lnss3 -lnssutil3 -lsmime3 -lssl3 -lplds4 -lplc4 -lnspr4
checking for NSS_Init... no
checking for nss.h... yes
checking for ssl.h... no
configure: error: Mozilla NSS not found (required for SSL support)
pi@raspberrypi:~/nut $ make
make: *** No targets specified and no makefile found.  Stop.
pi@raspberrypi:~/nut $

--- a/drivers/riello_ser.c
+++ b/drivers/riello_ser.c
@@ -43,7 +43,10 @@
 #include "riello.h"
 
 #define DRIVER_NAME	"Riello serial driver"
-#define DRIVER_VERSION	"0.04"
+#define DRIVER_VERSION	"0.07"
+
+#define DEFAULT_OFFDELAY   5
+#define DEFAULT_BOOTDELAY  5
 
 /* driver description structure */
 upsdrv_info_t upsdrv_info = {
@@ -62,6 +65,9 @@ static uint8_t typeRielloProtocol;
 
 static uint8_t input_monophase;
 static uint8_t output_monophase;
+
+static unsigned int offdelay = DEFAULT_OFFDELAY;
+static unsigned int bootdelay = DEFAULT_BOOTDELAY;
 
 static TRielloData DevData;
 
@@ -761,6 +767,14 @@ void upsdrv_initinfo(void)
 	dstate_addcmd("shutdown.stop");
 	dstate_addcmd("test.battery.start");
 
+	dstate_setinfo("ups.delay.shutdown", "%u", offdelay);
+	dstate_setflags("ups.delay.shutdown", ST_FLAG_RW | ST_FLAG_STRING);
+	dstate_setaux("ups.delay.shutdown", 3);
+	dstate_setinfo("ups.delay.reboot", "%u", bootdelay);
+	dstate_setflags("ups.delay.reboot", ST_FLAG_RW | ST_FLAG_STRING);
+	dstate_setaux("ups.delay.reboot", 3);
+
+
 	if (typeRielloProtocol == DEV_RIELLOGPSER)
 		dstate_addcmd("test.panel.start");
 
@@ -813,9 +827,14 @@ void upsdrv_updateinfo(void)
 	dstate_setinfo("input.bypass.frequency", "%.2f", DevData.Fbypass/10.0);
 	dstate_setinfo("output.frequency", "%.2f", DevData.Fout/10.0);
 	dstate_setinfo("battery.voltage", "%.1f", DevData.Ubat/10.0);
-	dstate_setinfo("battery.charge", "%u", DevData.BatCap);
-	dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
-	dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
+
+	if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+		dstate_setinfo("battery.charge", "%u", DevData.BatCap);
+		dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
+	}
+
+	if (DevData.Tsystem < 0xFF)
+		dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
 
 	if (input_monophase) {
 		dstate_setinfo("input.voltage", "%u", DevData.Uinp1);

--- a/drivers/riello_usb.c
+++ b/drivers/riello_usb.c
@@ -979,10 +979,10 @@ void upsdrv_initinfo(void)
 	dstate_addcmd("test.battery.start");
 	dstate_addcmd("test.panel.start");
 
-	dstate_setinfo("ups.delay.shutdown", "%d", offdelay);
+	dstate_setinfo("ups.delay.shutdown", "%u", offdelay);
 	dstate_setflags("ups.delay.shutdown", ST_FLAG_RW | ST_FLAG_STRING);
 	dstate_setaux("ups.delay.shutdown", 3);
-	dstate_setinfo("ups.delay.reboot", "%d", bootdelay);
+	dstate_setinfo("ups.delay.reboot", "%u", bootdelay);
 	dstate_setflags("ups.delay.reboot", ST_FLAG_RW | ST_FLAG_STRING);
 	dstate_setaux("ups.delay.reboot", 3);
 
@@ -1076,6 +1076,7 @@ void upsdrv_updateinfo(void)
 		dstate_setinfo("battery.charge", "%u", DevData.BatCap);
 		dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
 	}
+
 	if (DevData.Tsystem < 0xFF)
 		dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
 

--- a/drivers/riello_usb.c
+++ b/drivers/riello_usb.c
@@ -36,7 +36,10 @@
 #include "riello.h"
 
 #define DRIVER_NAME	"Riello USB driver"
-#define DRIVER_VERSION	"0.06"
+#define DRIVER_VERSION	"0.07"
+
+#define DEFAULT_OFFDELAY   5  /*!< seconds (max 0xFF) */
+#define DEFAULT_BOOTDELAY  5  /*!< seconds (max 0xFF) */
 
 /* driver description structure */
 upsdrv_info_t upsdrv_info = {
@@ -54,6 +57,10 @@ static uint8_t gpser_error_control;
 
 static uint8_t input_monophase;
 static uint8_t output_monophase;
+
+/*! Time in seconds to delay before shutting down. */
+static unsigned int offdelay = DEFAULT_OFFDELAY;
+static unsigned int bootdelay = DEFAULT_BOOTDELAY;
 
 static TRielloData DevData;
 
@@ -972,6 +979,13 @@ void upsdrv_initinfo(void)
 	dstate_addcmd("test.battery.start");
 	dstate_addcmd("test.panel.start");
 
+	dstate_setinfo("ups.delay.shutdown", "%d", offdelay);
+	dstate_setflags("ups.delay.shutdown", ST_FLAG_RW | ST_FLAG_STRING);
+	dstate_setaux("ups.delay.shutdown", 3);
+	dstate_setinfo("ups.delay.reboot", "%d", bootdelay);
+	dstate_setflags("ups.delay.reboot", ST_FLAG_RW | ST_FLAG_STRING);
+	dstate_setaux("ups.delay.reboot", 3);
+
 	/* install handlers */
 /*	upsh.setvar = hid_set_value; setvar; */
 
@@ -1058,9 +1072,13 @@ void upsdrv_updateinfo(void)
 	dstate_setinfo("input.bypass.frequency", "%.2f", DevData.Fbypass/10.0);
 	dstate_setinfo("output.frequency", "%.2f", DevData.Fout/10.0);
 	dstate_setinfo("battery.voltage", "%.1f", DevData.Ubat/10.0);
-	dstate_setinfo("battery.charge", "%u", DevData.BatCap);
-	dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
-	dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
+	if ((DevData.BatCap < 0xFFFF) &&  (DevData.BatTime < 0xFFFF)) {
+		dstate_setinfo("battery.charge", "%u", DevData.BatCap);
+		dstate_setinfo("battery.runtime", "%u", DevData.BatTime*60);
+	}
+	if (DevData.Tsystem < 0xFF)
+		dstate_setinfo("ups.temperature", "%u", DevData.Tsystem);
+
 
 	if (input_monophase) {
 		dstate_setinfo("input.voltage", "%u", DevData.Uinp1);

